### PR TITLE
AP_HAL:examples:Printf: Improvements in the Printf  example

### PR DESCRIPTION
**Pull request by arshPratap**
Added improvements to the Printf example sketch as suggested [here](https://ardupilot.org/dev/docs/learning-ardupilot-the-example-sketches.html#rough-example-code). Thank you

### CODE REVIEW

**Review by khancyr — COMMENTED** [review]
hello, thanks for this, but I don't think we want this level of comment on each example. We expect people to have some coding knowledge before trying to try to modify ArduPilot codebase

File: libraries/AP_HAL/examples/Printf/Printf.cpp [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
-const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();    //create a reference to AP_HAL::HAL object to get access to hardware specific functions. For more info see <https://ardupilot.org/dev/docs/learning-ardupilot-the-example-sketches.html/>
 
+// the setup function runs once when the board powers up
 void setup(void) {
-    hal.console->printf("Starting Printf test\n");
+    hal.console->printf("Starting Printf test\n");    //print a starting message
 }
 
+//create a struct array float_tests
```
Comment (khancyr) [inline libraries/AP_HAL/examples/Printf/Printf.cpp:31]: comment need a space after //

Comment (arshPratap) [reply libraries/AP_HAL/examples/Printf/Printf.cpp:31]: Thank you, Sir, for the suggestion. I will work on the changes

**Comment by arshPratap** [comment]
> hello, thanks for this, but I don't think we want this level of comment on each example. We expect people to have some coding knowledge before trying to try to modify ArduPilot codebase

@khancyr thank you for pointing that out sir. Actually yesterday I found that some of the library examples were poorly commented and some of them did not even have proper comments and I was planning to rectify them, but now even I think having a high comment count per example is not a good idea. Will a precise number of comments be good for this PR and some future library example related commits?   Thanks again!!!

**Comment by khancyr** [comment]
Hello,

sorry about my previous message. I have speak with the other and it is fine. 
On a general rule on comment, we don't need comment on each line or variable. They should be for explaining some logic or unusual variable/pattern. 
For example in this PR, you comment each printf method and `//uinsigned 8 bit integer to work as an iterator` that is pretty obvious if you have a little base on coding.

**Comment by arshPratap** [comment]
Thank you, Sir, for the reply!! I will keep this in mind for future commits and will make the changes.

**Comment by arshPratap** [comment]
@khancyr  sir, I have made the necessary changes. Please take a look

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**